### PR TITLE
Delete files after finish the job

### PR DESCRIPTION
**Pull request by fabyan-alexander**
* Instead of loading the data of all files by a pattern, we'll get the list of files of the oldest day and then load their data.
* Delete the loaded files when the job is completed.
* Add support for the legacy timestamp on file names.

### CODE REVIEW

**Comment by fabyan-alexander** [comment]
[[BillingAg] Delete parquet files used to generate the aggregated data](https://trello.com/c/gEdzInFX/983-billingag-delete-parquet-files-used-to-generate-the-aggregated-data)

File: src/main/java/com/leaf/billing/utils/FileUtils.java [hunk 1]
```diff
@@ -14,7 +15,11 @@ private FileUtils() {
     public static LocalDate extractTimestampFromFileName(String path) {
         String name = FileNameUtils.getBaseName(path).split("_", 2)[1];
         DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH-mm-ss");
-        return formatter.parse(name, LocalDate::from);
+        try {
+            return formatter.parse(name, LocalDate::from);
+        } catch (DateTimeParseException e) {
+            return DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss").parse(name.substring(0, 19), LocalDate::from);
```
Comment (ricardosantacattarina) [inline src/main/java/com/leaf/billing/utils/FileUtils.java:21]: what a weird pattern huh :eyes:

**Review by ricardosantacattarina — APPROVED** [review]
Nice! but I couldn't find anything to suggest though

**Review by rgdafonso — APPROVED** [review]
Let's gamble. Try merging.

File: src/main/java/com/leaf/billing/utils/FileUtils.java [hunk 2]
```diff
@@ -14,7 +15,11 @@ private FileUtils() {
...
+        }
```
[the remark this answers is not in the document]
Comment (fabyan-alexander) [reply src/main/java/com/leaf/billing/utils/FileUtils.java:22]: There are different patterns in the source folder (the ones before you fixed the timestamp)

File: src/main/java/com/leaf/billing/FieldBoundaryAggregatorApp.java [hunk 3]
```diff
@@ -37,37 +37,37 @@ public static void main(String[] args) throws IOException {
...
-                date_trunc("day", new Column("timestamp")));
+        String sourceFolder = jobProperties.getSourceBasePath();
+        List<String> oldestFilesByDay = SparkCommands.getOldestFilesByDay(sourceFolder);
+        if (oldestFilesByDay.isEmpty()) {
+            log.info("There is no files in sourceFolder {}", sourceFolder);
+            return 0;
+        }
+        Dataset<Row> aggregated = session.read().format("parquet").load(oldestFilesByDay.toArray(new String[0]));
+        Dataset<Row> unique = dropDuplicatedColumns(aggregated)
+                .withColumn("yearMonthDayTimestamp", date_trunc("day", new Column("timestamp")));
```
[the remark this answers is not in the document]
Comment (fabyan-alexander) [reply src/main/java/com/leaf/billing/FieldBoundaryAggregatorApp.java:50]: The name of the column inside the event is timestamp :thinking:

**Pull request merged**